### PR TITLE
Cog 1090 view details from change log

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
#### Which User story does this relate to (link)?
[COG-1090, changeLog details view](https://osi-cwds.atlassian.net/browse/COG-1090)

#### Brief feature description
- View changeLog details from the changeLog table.

### CODE REVIEW

**Comment by Sushmareddy-OSI** [comment]
retest this please

File: app/javascript/views/userDetail/UserDetail.jsx [hunk 1]
```diff
@@ -165,7 +165,7 @@ export default class UserDetail extends Component {
               <CWSPermissions />
             </div>
             <div className="col-md-8">
-              <ChangeLog />
+              <ChangeLog permissionsList={possiblePermissionsList} rolesList={possibleRolesList} />
```
Comment (RamuKammagani-osi) [inline app/javascript/views/userDetail/UserDetail.jsx]: Can you add test case for this?

File: app/javascript/_utils/formatters.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* eslint camelcase: 0 */
 
 import { DateTime } from 'luxon'
+import { userStatusFormat } from '../_constants/constants'
```
Comment (RamuKammagani-osi) [inline app/javascript/_utils/formatters.js]: Can you rename this function to accountStatusFormat as this formats boolean to active/inactive?

File: app/javascript/_utils/formatters.test.js [hunk 3]
```diff
@@ -140,31 +148,81 @@ describe('#checkDate', () => {
...
+    expect(formatChangeLogValues(eventType, value, permissionslist, rolesList)).toEqual('roleOne')
+  })
+
+  it('returns the status value when eventType is Account Status', () => {
+    const eventType = 'Account Status'
+    const value = true
+    expect(formatChangeLogValues(eventType, value, permissionslist, rolesList)).toEqual('Active')
+  })
+
+  it('returns empty string when there is no eventType', () => {
```
Comment (Nasir2712) [inline app/javascript/_utils/formatters.test.js]: also maybe a scenario if eventype is null?

File: app/javascript/selectors/detailSelector.test.js [hunk 4]
```diff
@@ -228,35 +228,29 @@ describe('selectors', () => {
...
     })
 
-    it('renders false when given empty array ', () => {
-      const assignedPermissions = []
-      const state = getState({
-        assignedPermissions: assignedPermissions,
-        permissionsList: permissionsList,
+    describe('When resent assigned Roles is undefined ', () => {
+      it('returns just empty', () => {
+        const state = getState({ assignedPermissions: undefined, permissionsList })
```
Comment (Nasir2712) [inline app/javascript/selectors/detailSelector.test.js]: one more scenario if assignedPermissions and permissionsList are null?

File: app/javascript/views/userDetail/ChangeLog.jsx [hunk 5]
```diff
@@ -62,7 +80,36 @@ const ChangeLog = () => (
...
+              accessor: 'admin_name',
+              minWidth: 70,
+            },
+            {
+              Header: 'Notes & Details',
+              // eslint-disable-next-line react/display-name
+              Cell: ({ row, original }) => (
+                <ModalComponent
+                  data={row}
+                  oldValue={formatChangeLogValues(row.event_type, original.old_value, permissionsList, rolesList)}
```
Comment (Nasir2712) [inline app/javascript/views/userDetail/ChangeLog.jsx]: this is interesting, how does the row know which event_type to send? am I missing something?

File: app/javascript/_utils/formatters.js [hunk 6]
```diff
@@ -57,3 +58,29 @@ export function formatSelectedRoles(assignedRoles, rolesList) {
...
+  switch (eventType) {
+    case 'Permission':
+      return formatPermissions(value, permissionsList)
+    case 'User Role':
+      return formatRoles(value, rolesList)
+    case 'Account Status':
+      return accountStatusFormat({ enabled: value })
+    default:
+      return ''
+  }
```
[the remark this answers is not in the document]
Comment (Sushmareddy-OSI) [reply app/javascript/_utils/formatters.js]: There is nothing to format those values, 
Email Address Updates, no need to format the email.

For User Creation, Registration Completion, Registration Resends eventTypes, there is no strategy decided yet to display that information in Modal.

File: app/javascript/_utils/formatters.js [hunk 7]
```diff
@@ -57,3 +58,29 @@ export function formatSelectedRoles(assignedRoles, rolesList) {
...
+      .map(permission => permissionList.find(d => d.value === permission))
+      .filter(value => Boolean(value))
+      .map(({ value, label }) => label)
+      .join(', ')
+  )
+}
+
+export function formatChangeLogValues(eventType, value, permissionsList, rolesList) {
+  switch (eventType) {
+    case 'Permission':
```
Comment (kelfink) [inline app/javascript/_utils/formatters.js]: are these case values expected to come straight from the backend?  Maybe the default should put out a console warning that the type wasn't understood.

File: app/javascript/_utils/formatters.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (sunilb0575) [reply app/javascript/_utils/formatters.js]: ok, but I think we should at least show the  email address change. Right? Like showing old email and new email on the modal.

File: app/javascript/_utils/formatters.js (hunk 7 above)
Comment (Sushmareddy-OSI) [reply app/javascript/_utils/formatters.js]: Yes, these are expected from backend.

File: app/javascript/_utils/formatters.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Sushmareddy-OSI) [reply app/javascript/_utils/formatters.js]: thanks @sunilb0575 , updated!

File: app/javascript/_utils/formatters.test.js (hunk 3 above)
Comment (Sushmareddy-OSI) [reply app/javascript/_utils/formatters.test.js]: according to the formatter method(formatChangeLogValues), the default case will return value (old_value or new_value) if the eventType is not recognized

**Pull request merged**